backend/src/database/models.rs

```diff
@@ -390,16 +389,37 @@ pub struct Role {
     pub updated_at: NaiveDateTime,
 }
 
-#[derive(Insertable)]
+#[derive(FromForm)]
+pub struct RoleUpdateInput {
+    pub name: String,
+    pub description: Option<String>,
+    pub min_available: i32,
+    pub max_available: i32,
+    pub finalised: bool,
+}
+
+#[derive(Insertable, AsChangeset)]
 #[table_name = "roles"]
-pub struct NewRole {
+pub struct NewOrUpdateRole {
     pub name: String,
     pub description: Option<String>,
     pub min_available: i32,
     pub max_available: i32,
     pub finalised: bool,
 }
 
+impl NewOrUpdateRole {
+    pub fn from(role_input: &RoleUpdateInput) -> NewOrUpdateRole {
```

Comment:
I think Zac meant to impl the `std::convert::From` trait - ie

`impl From<&RoleUpdateInput> for NewOrUpdateRole`

https://doc.rust-lang.org/std/convert/trait.From.html
https://stevedonovan.github.io/rustifications/2018/09/08/common-rust-traits.html (see From and Into)